python-backend/app/api/mines/reports/resources/mine_reports.py

```diff
@@ -47,7 +46,12 @@ class MineReportListResource(Resource, UserMixin):
     @api.doc(description='returns the reports for a given mine.')
     @requires_role_view_all
     def get(self, mine_guid):
-        mine_reports = MineReport.find_by_mine_guid(mine_guid)
+        mrd_category = request.args.get('mine_report_definition_category')
+        if mrd_category:
+            mine_reports = MineReport.find_by_mine_guid_with_category(mine_guid, mrd_category)
```

Comment:
Maybe nitpicky but is mine_guid_and_category more clearer than mine_guid_with_category? Just up for thought

Reply:
i like your name.